sdk/src/main/java/software/amazon/awssdk/iot/AwsIotMqttConnectionBuilder.java

```diff
@@ -360,6 +360,9 @@ public AwsIotMqttConnectionBuilder withProtocolOperationTimeoutMs(int protocolOp
      * @return {@link AwsIotMqttConnectionBuilder}
      */
     public AwsIotMqttConnectionBuilder withTimeoutMs(int timeoutMs) {
```

Comment:
Is it possible for this to be called before `withSocketOptions()` is called? If so, it's possible a customer may set `withTimeoutMs()` using this which is then overridden by the `withSocketOptions()`. Maybe we should also add a null check to `withSocketOptions()` and if it's not null, copy over the timeout from it to the newly set socket options. Or that may be overcomplicating things and we just allow the timeout to be overwritten if the customer somehow decides to use both.